### PR TITLE
Move Font*, Image* and Margins converters to System.Drawing.Common and apply Unit and Name converter

### LINKED ISSUE #1011: System.Drawing.ImageConverter raise the System.PlatformNotSupportedException on Linux
dotnet --version
3.1.100
nuget packeges:
- System.Drawing.Common  4.7.0
- System.Configuration.ConfigurationManager 4.7.0
OS: Linux

**Steps to reproduce:**
Environment: Linux with dotnet 3.1.100

1) dotnet new console
2) dotnet add package System.Drawing.Common   
Installed version: 4.7.0
3) modify Program.cs file 
add  new lines:
`var converter = System.ComponentModel.TypeDescriptor.GetConverter(typeof(System.Drawing.Image));
Console.WriteLine($"{converter?.GetType()}");`
4) dotnet run
Console output: 
>"System.ComponentModel.TypeConverter"
5)  dotnet add package System.Configuration.ConfigurationManager
Installed version: 4.7.0
6) dotnet run
Console output: 
> Unhandled exception. System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.PlatformNotSupportedException: System.Windows.Extensions types are not supported on this platform.
   at System.Drawing.ImageConverter..ctor()
   --- End of inner exception stack trace ---
   at System.RuntimeTypeHandle.CreateInstance(RuntimeType type, Boolean publicOnly, Boolean wrapExceptions, Boolean& canBeCached, RuntimeMethodHandleInternal& ctor, Boolean& hasNoDefaultCtor)
   at System.RuntimeType.CreateInstanceDefaultCtorSlow(Boolean publicOnly, Boolean wrapExceptions, Boolean fillCache)
   at System.RuntimeType.CreateInstanceDefaultCtor(Boolean publicOnly, Boolean skipCheckThis, Boolean fillCache, Boolean wrapExceptions)
   at System.Activator.CreateInstance(Type type, Boolean nonPublic, Boolean wrapExceptions)
   at System.Activator.CreateInstance(Type type)
   at System.ComponentModel.ReflectTypeDescriptionProvider.CreateInstance(Type objectType, Type callingType)
   at System.ComponentModel.ReflectTypeDescriptionProvider.ReflectedTypeData.GetConverter(Object instance)
   at System.ComponentModel.TypeDescriptor.TypeDescriptionNode.DefaultTypeDescriptor.System.ComponentModel.ICustomTypeDescriptor.GetConverter()
   at System.ComponentModel.TypeDescriptor.GetConverter(Type type)

### LINKED ISSUE #33061: System.Drawing.Font Name and Unit properties are missing TypeConverterAttributes
Both the `Font.Name` and `Font.Unit` properties are missing attributes for their respective type converters, `FontNameConverter` and `FontUnitConverter`. Both of these are included in `System.Windows.Extensions`. These missing type converters negatively affect how the font properties are displayed in the property grid control.

### PR DESCRIPTION
**Pull request by safern**
Fixes: https://github.com/dotnet/runtime/issues/33061
Fixes: https://github.com/dotnet/runtime/issues/1011

cc: @DustinCampbell @merriemcgaw @RussKie 

@Anipik I'm using the old Generate PNSE exclude API list attribute here.

### CODE REVIEW

**Comment by safern** [comment]
Touching ref assemblies but not adding new API.

**Comment by eerhardt** [comment]
>     <TargetFrameworks>$(NetCoreAppCurrent);netcoreapp3.0</TargetFrameworks>

This project still builds for netcoreapp3.0. Do we need to handle this specially?

---
Refers to: src/libraries/System.Windows.Extensions/ref/System.Windows.Extensions.csproj:3 in 8554791. [](commit_id = 85547916ea72b8bf8993c85028e5283c44163ddd, deletion_comment = False)

File: src/libraries/System.Windows.Extensions/src/System.Windows.Extensions.csproj [hunk 1]
```diff
@@ -40,23 +46,16 @@
...
-    <Compile Include="System\Drawing\Printing\MarginsConverter.cs" />
     <Compile Include="System\Media\SoundPlayer.cs" />
     <Compile Include="System\Media\SystemSound.cs" />
     <Compile Include="System\Media\SystemSounds.cs" />
     <Compile Include="System\Xaml\Permissions\XamlAccessLevel.cs" />
   </ItemGroup>
-  <ItemGroup Condition="'$(TargetsWindows)' == 'true'">
-    <ProjectReference Include="$(LibrariesProjectRoot)System.Drawing.Common\src\System.Drawing.Common.csproj" />
+  <ItemGroup>
+    <ProjectReference Include="$(LibrariesProjectRoot)System.Drawing.Common\ref\System.Drawing.Common.csproj" />
```
Comment (eerhardt) [inline src/libraries/System.Windows.Extensions/src/System.Windows.Extensions.csproj:58]: When do we decide to P2P to the `ref` and when do we use the `src`?

File: src/libraries/System.Drawing.Common/src/System.Drawing.Common.csproj [hunk 2]
```diff
@@ -116,6 +116,12 @@
     <Compile Include="System\Drawing\Internal\ISystemEventTracker.cs" />
     <Compile Include="System\Drawing\Brush.cs" />
     <Compile Include="System\Drawing\Font.cs" />
+    <!-- TODO sort -->
```
Comment (eerhardt) [inline src/libraries/System.Drawing.Common/src/System.Drawing.Common.csproj:119]: Are you going to fix this TODO in this PR or some other time?

File: src/libraries/System.Drawing.Common/ref/System.Drawing.Common.cs [hunk 3]
```diff
@@ -961,7 +1012,7 @@ public sealed partial class Region : System.MarshalByRefObject, System.IDisposab
         public bool IsVisible(System.Drawing.Rectangle rect, System.Drawing.Graphics? g) { throw null; }
         public bool IsVisible(System.Drawing.RectangleF rect) { throw null; }
         public bool IsVisible(System.Drawing.RectangleF rect, System.Drawing.Graphics? g) { throw null; }
-        public bool IsVisible(int x, int y, System.Drawing.Graphics g) { throw null; }
+        public bool IsVisible(int x, int y, System.Drawing.Graphics? g) { throw null; }
```
Comment (eerhardt) [inline src/libraries/System.Drawing.Common/ref/System.Drawing.Common.cs:1015]: Intentional change?

File: src/libraries/System.Drawing.Common/src/System.Drawing.Common.csproj (hunk 2 above)
Comment (safern) [reply src/libraries/System.Drawing.Common/src/System.Drawing.Common.csproj:119]: Ah good catch? The TODO was to actually introduced them respecting the alphabetical ordering.

File: src/libraries/System.Windows.Extensions/src/System.Windows.Extensions.csproj (hunk 1 above)
Comment (safern) [reply src/libraries/System.Windows.Extensions/src/System.Windows.Extensions.csproj:58]: Right now we just need the reference to the converters we're typeforwarding we no longer need anything from the implementation.

File: src/libraries/System.Drawing.Common/ref/System.Drawing.Common.cs (hunk 3 above)
Comment (safern) [reply src/libraries/System.Drawing.Common/ref/System.Drawing.Common.cs:1015]: This was from running GenerateReferenceSource it seems like we missed this nullable annotation in the ref.

**Comment by safern** [comment]
> This project still builds for netcoreapp3.0. Do we need to handle this specially?

Not really because it is not part of the shared framework and the typeforwards are yo System.Drawing.Common which will be in the netcoreapp3.0 implementation as well which is not inbox either, so it will have a PackageReference to System.Drawing.Common so it would work fine.

**Review by eerhardt — APPROVED** [review]

**Pull request merged**